### PR TITLE
Removed variants deprecations

### LINKED ISSUE #6166: Variants deprecation changes
Because of #5972 deprecations needs to be handled (search code for `issue:6166`):
- [x] remove the `tap` in `storefrontlib/src/cms-components/product/product-images/product-images.component.ts`
- [x] remove `isThumbsEmpty` in the same file.

Initially this ticet was just for the above part. But currently we basically need to check all deprecated/todo code related to variants; see master ticket: #6306

### PR DESCRIPTION
**Pull request by mirkosoft**
closes #6166

### CODE REVIEW

File: projects/storefrontlib/src/cms-components/myaccount/order/return-request-detail/return-request-items/return-request-items.component.html [hunk 1]
```diff
@@ -42,19 +42,7 @@
...
-                      .variantOptionQualifiers
-                  "
-                  class="cx-property"
-                >
-                  <div class="cx-label">{{ variant.name }}</div>
-                  <div class="cx-value">{{ variant.value }}</div>
-                </div>
-              </ng-container>
-              <ng-container *cxFeatureLevel="'1.5'">
+              <ng-container>
```
Comment (dunqan) [inline projects/storefrontlib/src/cms-components/myaccount/order/return-request-detail/return-request-items/return-request-items.component.html]: do you need that `ng-container`?

File: projects/storefrontlib/src/cms-components/product/product-images/product-images.component.ts [hunk 2]
```diff
@@ -22,19 +22,8 @@ export class ProductImagesComponent {
     )
   );
 
-  /**
-   * @deprecated since version 1.5
-   * This variable will no longer be in use. Use thumbs$ observable instead.
-   * TODO(issue:#6166).
-   */
-  isThumbsEmpty: boolean;
```
Comment (znikola) [inline projects/storefrontlib/src/cms-components/product/product-images/product-images.component.ts:30]: This should be handled in update-schematics in two ways:
1. if customers are using the selector for this component, we should put a comment in their html template saying that this property has been removed. To do this, please follow the README instructions defined here https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/6769 (not yet merged)
2. if customers extended this component and use `isThumbsEmpty`, they'll get an error. In this case we should also put a comment in their ts file. To do this, first you need to check if the customer's component extends our component - `projects/schematics/src/shared/utils/file-utils.ts#checkInheritance`, and if it does then you can use `projects/schematics/src/shared/utils/file-utils.ts#insertCommentAboveIdentifier`, but please wait for this PR https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/6790, as it will bring some required changes.

File: projects/schematics/src/migrations/2_0/component-deprecations/component-deprecations.ts [hunk 3]
```diff
@@ -29,14 +29,16 @@ export function migrate(): Rule {
...
       for (const deprecatedComponent of COMPONENT_DEPRECATION_DATA) {
-        for (const removedProperty of deprecatedComponent.removedProperties) {
-          const change = insertHtmlComment(
-            content,
-            deprecatedComponent.selector,
-            removedProperty
-          );
-          if (change) {
-            tree.overwrite(htmlFilePath, change);
+        if (htmlFilePath.includes(deprecatedComponent.componentClassName)) {
```
Comment (znikola) [inline projects/schematics/src/migrations/2_0/component-deprecations/component-deprecations.ts]: Can you help me understand, why was this added?

File: projects/schematics/src/shared/utils/file-utils.ts [hunk 4]
```diff
@@ -122,37 +122,25 @@ export function insertHtmlComment(
...
+  // for cases like: <cx-consent-management-form isLevel13="xxx"></cx-consent-management-form>
+  const selectorRegExp = new RegExp(`(<${componentSelector}[^>]*>)`, 'gi');
+  content = content.replace(selectorRegExp, `${comment}\$1`);
 
-    newContent = newContent.slice(0, index) + comment + newContent.slice(index);
-    index += comment.length + componentSelector.length;
+  // for cases like: <div *ngIf="isThumbsEmpty">test</div>
+  if (newContent === content) {
+    // content hasn't been changed by the above selector regexp
+    const propertyRegExp = new RegExp(`(<.+${componentProperty})`, 'g');
```
Comment (znikola) [inline projects/schematics/src/shared/utils/file-utils.ts]: Did you mean `componentProperty.name`?

File: projects/schematics/src/shared/utils/file-utils.ts [hunk 5]
```diff
@@ -122,37 +122,25 @@ export function insertHtmlComment(
...
+    newContent = content.replace(propertyRegExp, `${comment}\$1`);
+  } else {
+    newContent = content;
```
Comment (znikola) [inline projects/schematics/src/shared/utils/file-utils.ts]: Do we need this last `else`?

File: projects/schematics/src/shared/utils/file-utils.ts [hunk 6]
```diff
@@ -122,37 +122,25 @@ export function insertHtmlComment(
   componentSelector: string,
   componentProperty: ComponentProperty
 ): string | undefined {
-  const selector = buildSelector(componentSelector);
+  let newContent = content;
```
Comment (znikola) [inline projects/schematics/src/shared/utils/file-utils.ts]: We should add tests that cover the `*ngIf` cases

**Review by znikola — CHANGES_REQUESTED** [review]

File: projects/schematics/src/shared/utils/file-utils.ts (hunk 5 above)
Comment (mirkosoft) [reply projects/schematics/src/shared/utils/file-utils.ts]: nope, changed this condition.

File: projects/schematics/src/shared/utils/file-utils.ts (hunk 4 above)
Comment (mirkosoft) [reply projects/schematics/src/shared/utils/file-utils.ts]: yes, thanks.

File: projects/schematics/src/shared/utils/file-utils.ts (hunk 6 above)
Comment (mirkosoft) [reply projects/schematics/src/shared/utils/file-utils.ts]: just added it

File: projects/schematics/src/migrations/2_0/component-deprecations/component-deprecations.ts (hunk 3 above)
Comment (mirkosoft) [reply projects/schematics/src/migrations/2_0/component-deprecations/component-deprecations.ts]: fix pushed

**Review by znikola — DISMISSED** [review]
Schematics part looks good to me

**Review by dunqan — APPROVED** [review]

**Review by dunqan — APPROVED** [review]
